File: go/codekatas/FizzBuzz.go
package codekatas

import (
	   "strconv"
	   "strings"
	   )

func FizzBuzzSingle(number int) string {
	 switch {
	 	case number%3 == 0 && number%5 == 0:
			 return "FizzBuzz"
		case number%3 == 0:
			 return "Fizz"
		case number%5 == 0:
			 return "Buzz"
	 }
	 
	 return strconv.Itoa(number)
}

func FizzBuzz(number int) string {
	 temp := []string{}
	 for counter := 1; counter <= number; counter ++ {
	 	 temp = append(temp, FizzBuzzSingle(counter))
	 }

	 return strings.Join(temp, " " ) 
}

File: csharp/katasAlgorithms/FizzBuzz.cs
using System;
using System.Collections.Generic;

namespace katasAlgorithms
{
    public class FizzBuzz
    {
        public bool DivisibleBy5(int number)
        {
            return number % 5 == 0;
        }

        public bool DivisibleBy3(int number)
        {
            return number % 3 == 0;
        }

        public string ConvertSingle(int number)
        {
            if (DivisibleBy3(number) && DivisibleBy5(number))
            {
                return "FizzBuzz";
            }

            if (DivisibleBy3(number))
            {
                return "Fizz";
            }

            if (DivisibleBy5(number))
            {
                return "Buzz";
            }

            return number.ToString();
        }

        public string Run(int numberOfIterations)
        {
            List<string> results = new List<string>();
            for (int counter = 1; counter <= numberOfIterations; counter++)
            {
                results.Add(ConvertSingle(counter));
            }

            return String.Join(" ", results);
        }
    }
}

